### PR TITLE
zfssa-3: update to k8s 1.25.7

### LINKED ISSUE #3: Implement CSI Volume Cloning Feature
The initial version of the driver does not implement volume cloning (https://v1-20.docs.kubernetes.io/docs/concepts/storage/volume-pvc-datasource/).

This should be achievable by
- creating a snapshot of the PVC referenced
- create the clone
- export the clone as a file system or LUN

This issue is 'generic' and may be split up if the filesystem path is substantially different from iscsi. For now, filesystem would be the priority since it would be common for boot images and home directories and such.

### PR DESCRIPTION
**Pull request by pmonday**
This branch morphed into, basically, a k8s update to get us back on track
* Update to k8s 1.25.7
* Update dependencies in project
* Fix csi-sanity failures that have crept in
* Re-shuffle examples and minor corrections throughout

Tested against
* Oracle ZFS Storage Appliance 8.8.x
* Marketplace image and hardware

The goal here is to step through 1.25.7 then come back to clone.

### CODE REVIEW

File: examples/nfs-volume-clone/README.md [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+for PVCs, allowing clients to use the same base filesystem for
+different volumes.
+
+Clones on an Oracle ZFS Storage Appliance start from a snapshot of the
+share, the snapshot is then cloned and made available. A snapshot takes
+no space as long as the origin filesystem is unchanged.
+
+More information about snapshots and clones on the Oracle ZFS Storage
+Appliance can be found in the
+[Oracle® ZFS Storage Appliance Administration Guide,](https://docs.oracle.com/cd/F13758_01/html/F13769/gprif.html)
```
Comment (jeongtae-kim) [inline examples/nfs-volume-clone/README.md:13]: FYI. It seems this is the external address: https://docs.oracle.com/en/storage/zfs-storage/zfs-appliance/os8-8-x/admin-guide/index.html, although the address above works too.

File: examples/nfs-volume-clone/README.md [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    zfssa-csi-nodeplugin-ph8qr     2/2     Running   0          8m32s
+    zfssa-csi-nodeplugin-wzgpq     2/2     Running   0          8m32s
+    zfssa-csi-provisioner-0        4/4     Running   0          8m32s
+    zfssa-nfs-volume-example-pod   1/1     Running   0          2m21s
+    ```
+
+## Write data to the volume
+
+Once the pod is deployed, write data to the volume in the pod:
+```yaml
```
Comment (jeongtae-kim) [inline examples/nfs-volume-clone/README.md:88]: text

File: examples/nfs-volume-clone/README.md [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+/mnt # ls
+/mnt # echo "hello world" > demo.txt
+/mnt # 
+```
+
+## Remove the example when complete
+
+Helm will remove all of our pods and data:
+```
+helm uninstall setup-volume
```
Comment (jeongtae-kim) [inline examples/nfs-volume-clone/README.md:154]: Shouldn't the order matter? I thought if you delete the original volume, all snapshots & clones from the volume are also deleted. If they are not deletable, deletion fails.

Comment (pmonday) [reply examples/nfs-volume-clone/README.md:154]: I'm going to remove clone from this branch, the branch will be version update and bug fixes only, I will address your comments in the clone branch when I re-create it. Thanks Jeongtae!

**Review by jeongtae-kim — APPROVED** [review]

**Pull request merged**